### PR TITLE
Add hover state for uncollpased accordion button

### PR DESCRIPTION
**Pull request by carkod**
## Done

Add hover state for uncollpased accordion button

Fixes https://github.com/canonical-web-and-design/vanilla-framework/issues/3950

## QA

- Open [demo](https://vanilla-framework-3973.demos.haus//docs/examples/patterns/accordion/default)  compare with [current](https://vanillaframework.io/docs/examples/patterns/accordion/default)

When accordion is opened, hover over the button to highlight the hover state. Now all buttons should have hover state

- Review updated documentation:
  - Updated existing accordion examples https://vanilla-framework-3973.demos.haus//docs/examples/patterns/accordion/default/docs/patterns/accordion#accordion
 

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] **Done in master already** Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical-web-and-design/vanilla-framework/blob/master/templates/docs/whats-new.md).
- [x] **Done in master already** Documentation side navigation should be updated with the relevant labels.

## Screenshots

hover-state

### CODE REVIEW

**Comment by petermakowski** [comment]
(I could be wrong as there's a lot of nesting and it's not obvious what the resulting CSS specificity will be just by looking at the PR.)

Could you double check that the hover is be applied correctly for other variants as well? For example `.p-accordion__tab--with-title`:

https://github.com/canonical-web-and-design/vanilla-framework/blob/8f6dab228cd98fbf1d2e9b9dac53903ffe733a17/scss/_patterns_accordion.scss#L105-L108

**Comment by carkod** [comment]
> (I could be wrong as there's a lot of nesting and it's not obvious what the resulting CSS specificity will be just by looking at the PR.)
> 
> Could you double check that the hover is be applied correctly for other variants as well? For example `.p-accordion__tab--with-title`:
> 
> https://github.com/canonical-web-and-design/vanilla-framework/blob/8f6dab228cd98fbf1d2e9b9dac53903ffe733a17/scss/_patterns_accordion.scss#L105-L108

Yes it works with all the other accordion variants:

https://vanilla-framework-3973.demos.haus/docs/examples/patterns/accordion/headings
https://vanilla-framework-3973.demos.haus/docs/examples/patterns/accordion/tick-elements

File: templates/docs/whats-new.md [hunk 1]
```diff
@@ -27,6 +27,12 @@ When we add, make significant updates, or deprecate a component we update their
       <td>2.36.0</td>
       <td>We've added a variation of the accordion using tick elements</td>
     </tr>
+    <tr>
```
Comment (bartaz) [inline templates/docs/whats-new.md:30]: No need for this, we are not adding new class name or anything. Just fixing a bug with a color.

**Review by bartaz — APPROVED** [review]
@carkod Just remove unnecessary mention in "what's new" and it's good to go, thanks!

**Pull request merged**